src/ros/mb_publisher.cpp

```diff
@@ -18,6 +18,36 @@ void MBPublisher::publish(ur_msgs::IOStates& io_msg, SharedMasterBoardData& data
   io_pub_.publish(io_msg);
 }
 
+void MBPublisher::publishRobotStatus(SharedRobotModeData& data)
+{
+  industrial_msgs::RobotStatus msg;
+  msg.drives_powered.val = data.robot_power_on;
+  msg.e_stopped.val = data.emergency_stopped;
+  msg.in_motion.val = data.program_running;
+  msg.error_code = 0;
+  msg.in_error.val = data.protective_stopped;
+  msg.mode.val = industrial_msgs::TriState::UNKNOWN;
+
+  status_pub_.publish(msg);
+}
+
+void MBPublisher::publishRobotStatus(RobotModeData_V3_0__1& data)
+{
+  industrial_msgs::RobotStatus msg;
+
+  msg.drives_powered.val = data.robot_power_on;
+  msg.e_stopped.val = data.emergency_stopped;
+  msg.in_motion.val = data.program_running;
+  msg.in_error.val = data.protective_stopped;
+  msg.motion_possible.val = (robot_mode_V3_X::RUNNING == data.robot_mode) ? industrial_msgs::TriState::ON : industrial_msgs::TriState::OFF;
```

Comment:
Nitpicking but `data.robot_mode == robot_mode_V3_X::RUNNING` (reverse ordering) is probably preferred as this is the style used elsewhere.

Reply:
ah, yes, sorry, I was experimenting if I could get this away:

    ur_modern_driver/src/ros/mb_publisher.cpp:41: Warnung: enumeral mismatch in conditional expression: ‘industrial_msgs::TriState_<std::allocator<void> >::<anonymous enum>’ vs ‘industrial_msgs::TriState_<std::allocator<void> >::<anonymous enum>’ [-Wenum-compare]
      msg.motion_possible.val = (data.robot_mode == robot_mode_V3_X::RUNNING) ? industrial_msgs::TriState::ON : industrial_msgs::TriState::OFF;
                                                                           ^